### DISCUSSION TITLE
Idea to avoid losing arming signal in flight

### DISCUSSION THREAD

**Discussion by DavidGom1** [iNavFlight/inav#9697] [Ideas]
INAV 6.0.1
Tx RADIOMASTER BOXER ELRS 3.3.0 FCC 1W dyn 250hz hybrid tlm:1/64
Rx BETAFPV SuperD ELRS 3.3.0 100mw

hello,
This weekend I was happily flying with friends, at a medium distance (+4km).
I suddenly lost the arming signal, disarming my fixed wing and causing it to fall, causing the fuselage to break.
I tried to reset, but it was not possible, I do not have any type of command in place to solve this, I will try it as a provisional way, although I will tell you an idea:

For pilots who use GPS, and we have variables of length to home, and relative altitude.
If you are more than 100m from home, and you are also more than 100m high, in order to disarm, it will be necessary to use a pre-arming button. It could even be the same button, create an option that would be a button for "pre-disarming"... or "pre-disarming in flight".
Furthermore, it would be very good if relative height and length variables to the house were modifiable.

They have recommended that I modify "small_angle = 180" via CLI, I suppose to be able to rearrange.

Finally, thank you for your great work, it brings a lot of fun to my weekends.
Great job

**Comment by MrD-RC** [comment]
There is an option to re-arm in flight for 5 seconds after disarming. This is in INAV 7.0. I suggest you upgrade. 

I don’t agree that it is ever a good idea to block disarming. The arm switch is a safety device. If a pilot is inconvenienced because they have accidentally disarmed (which we all have at some point). So be it. It’s better that that having someone injured because disarming was made more difficult. 

BTW. You can easily re-arm in flight with older versions of INAV too. Just lower the throttle, switch to manual, acro, angle, or horizon, let go of the sticks, arm. If you have pre-arm, you’ll need to do that too.
